src/storage/page_buffer.c

```diff
@@ -1950,8 +1950,7 @@ pgbuf_fix_release (THREAD_ENTRY * thread_p, const VPID * vpid, PAGE_FETCH_MODE f
 
   /* Set Page identifier if needed */
   // Redo recovery may find an immature page which should be set.
-  force_set_vpid = (fetch_mode == NEW_PAGE && log_is_in_crash_recovery_and_not_yet_completes_redo ());
```

Comment:
During recovery, if the page was fixed with a recovery index of `RCV_IS_NEW_PAGE_INIT`, `iopage.prv` was initialized forcefully. Could you tell us why now we can skip it?

Reply:
Thank you for clarifying and explaining in detail. It is very helpful. I also tried to find out the case it was needed, but I couldn't. 
Just so I'm clear, you mean that there is no case in which VPID of `prv` is not NULL, but `ptype` and `nonce` have to be initialized. And this is true even without this "is reserved check" patch, but you remove it during dealing with this patch.

Reply:
And I think the assertion you suggested seems good for safety because we are removing a case.

Reply:
I added the change on the else branch with asserts for `volid` and `pageid` and performed some tests.
As for checking `ptype` a higher level context info is needed and it is already done - see code after the call to `pgbuf_set_bcb_page_vpid` in caller functions `pgbuf_set_page_ptype` and `pgbuf_fix`.